### ISSUE TITLE
JavaProjectClasspathChangeAnalyzer misses cases

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#10]
`org.eclipse.xtext.ui.util.JavaProjectClasspathChangeAnalyzer.isRelevantPackageFragmentRootChange(IJavaElementDelta)`misses cases.
- if a jar on classpath changes in itself then the delta is `IJavaElementDelta.F_CONTENT | IJavaElementDelta.F_ARCHIVE_CONTENT_CHANGED`
- if the jar is inside the project `((IPackageFragmentRoot) element).isExternal()`is false

**Comment by cdietrich** [comment]
Here is a new testcase for `Bug486584Test`

```
@Test
	public void testFullBuildWhenClasspathChanged_2() throws CoreException {
		IJavaProject project = setupProject();
		IFile libaryFile = copyAndGetXtendExampleJar(project);
		IClasspathEntry libraryEntry = JavaCore.newLibraryEntry(libaryFile.getFullPath(), null, null);
		addToClasspath(project, libraryEntry);
		waitForBuild();
		assertFalse(getEvents().isEmpty());
		getEvents().clear();
		libaryFile.touch(null);
		libaryFile.refreshLocal(IResource.DEPTH_INFINITE, null);
		waitForBuild();
		assertFalse(getEvents().isEmpty());
		
	}
	
	@Test
	public void testFullBuildWhenClasspathChanged_3() throws CoreException, IOException {
		IJavaProject project = setupProject();
		IFile libaryFile = copyAndGetXtendExampleJar(project);
		IPath rawLocation = libaryFile.getRawLocation();
		IClasspathEntry libraryEntry = JavaCore.newLibraryEntry(rawLocation, null, null);
		addToClasspath(project, libraryEntry);
		waitForBuild();
		assertFalse(getEvents().isEmpty());
		getEvents().clear();
		File file = rawLocation.toFile();
		Files.touch(file);
		project.setRawClasspath(project.getRawClasspath(), null);
		project.getProject().getFile("src/dummy.txt").create(new StringInputStream(""), false, null);
		project.getProject().refreshLocal(IResource.DEPTH_INFINITE, null);
		waitForBuild();
		assertFalse(getEvents().isEmpty());
		
	}
```

**Comment by cdietrich** [comment]
And here a possible fix

```
public boolean isRelevantPackageFragmentRootChange(IJavaElementDelta delta) {
		IJavaElement element = delta.getElement();
		assert (element instanceof IPackageFragmentRoot);
		return delta.getKind() == IJavaElementDelta.REMOVED
			|| delta.getKind() == IJavaElementDelta.ADDED
			|| (delta.getFlags() & IJavaElementDelta.F_REORDER) != 0
			|| (delta.getFlags() & IJavaElementDelta.F_REMOVED_FROM_CLASSPATH) != 0
			|| (delta.getFlags() & IJavaElementDelta.F_ADDED_TO_CLASSPATH) != 0
			|| ((((IPackageFragmentRoot) element).isExternal() || ((IPackageFragmentRoot) element).isArchive()) && (delta.getFlags() & // external folders change
					(IJavaElementDelta.F_CONTENT | IJavaElementDelta.F_ARCHIVE_CONTENT_CHANGED)) == delta.getFlags());
	}
```

**Comment by cdietrich** [comment]
Help & Comments welcome

**Comment by cdietrich** [comment]
any news on this?

**Comment by cdietrich** [comment]
fixed in 2.12
